src/Build/Logging/SerialConsoleLogger.cs

```diff
@@ -488,6 +553,17 @@ public override void WarningHandler(object sender, BuildWarningEventArgs e)
             SetErrorsOrWarningsOnCurrentFrame();
             ShowDeferredMessages();
             setColor(ConsoleColor.Yellow);
+
+            //determine the mapping of properties to output
+            StringBuilder LogOutputProperties = new StringBuilder();
```

Comment:
you've repeated these few lines numerous times, can the duplication be eliminated?

Reply:
Yep I'll make a helper method and call that instead